File: src/main/java/com/dotgoing/model/Person.java
package com.dotgoing.model;

/**
 * Author: Sean
 * Date: 31/01/2018
 * Time: 3:04 PM
 */

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.extern.log4j.Log4j;

import javax.persistence.*;

@Data
@Log4j
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "person")
public class Person {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    private String firstName;
    private String lastName;
}

File: src/main/kotlin/com/dotgoing/model/Person.kt
package com.dotgoing.model

import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType.IDENTITY
import javax.persistence.Id
import javax.persistence.Table

@Entity
@Table(name = "person")
data class Person(var firstName: String = "", val lastName: String = "") {
    @Id
    @GeneratedValue(strategy = IDENTITY)
    val id: Long = 0
}